local.go

```diff
@@ -13,12 +13,18 @@ import (
 	"github.com/gonum/matrix/mat64"
 )
 
-// Local finds a local minimum of a function using a sequential algorithm.
-// In order to maximize a function, multiply the output by -1.
+// Local finds a local minimum of a minimization problem using a sequential
+// algorithm.
 //
-// The first argument is of Function type representing the function to be minimized.
-// Type switching is used to see if the function implements Gradient, FunctionGradient
-// and Statuser.
+// The first argument represents the problem to be minimized. Its fields are
+// functions that evaluate various quantities related to the problem like the
```

Comment:
How about it's fields are "routines"? It's obvious from the type that by routine we mean 'go function', and trying to minimize the number of different meanings of the word "function" is probably our own objective function :)